zerver/views/auth.py

```diff
@@ -326,6 +327,9 @@ def log_into_subdomain(request):
 
 def login_page(request, **kwargs):
     # type: (HttpRequest, **Any) -> HttpResponse
+    if is_subdomain_root_alias(request) and settings.REALMS_HAVE_SUBDOMAINS:
+        return HttpResponseNotFound("Login on root domain not allowed")
```

Comment:
We should probably just redirect them to the "find team" view rather than showing an error.